docs/design/openshift.md

```diff
@@ -191,4 +191,63 @@ $ cat release-manifests/image-references | head -n 10
 }
 ```
 
+## Update Image's Architecture ##
+
+The architecture of a release image is determined by the following algorithm:
+
+- check if it is a manifest list of length > 1 by [media-types](https://github.com/openshift/docker-distribution/blob/main/docs/spec/manifest-v2-2.md#media-types). 
+
+  Here `curl` and `skopeo` are chosen to show the information needed for the algorithm. It is up to the Cincinnati implementation for OpenShift how the information is retrieved. We may use `oc` (`oc image info` and `oc adm release info` to be precise) to achieve the same goal, but it would requires the understanding how `oc` gets the information. With `curl` and `skopeo` (as some oc commands communicate with an OpenShift cluster), it is clear that all the information comes from the image.
+
+  ```console
+  ### manifest list quay.io/openshift-release-dev/ocp-release:4.15.5-multi
+  $ curl -sH 'Accept: application/vnd.docker.distribution.manifest.list.v2+json' https://quay.io/v2/openshift-release-dev/ocp-release/manifests/4.15.5-multi | jq '.manifests|length'
+  4
+
+  ### manifest list for quay.io/openshift-release-dev/ocp-release:4.15.5-s390x
+  $ curl -sH 'Accept: application/vnd.docker.distribution.manifest.list.v2+json' https://quay.io/v2/openshift-release-dev/ocp-release/manifests/4.15.5-s390x | jq '.manifests|length'
```

Comment:
Instead of checking `manifests`, it might be enough to say "I asked for a manifest-list, but the registry just gave me a single manifest instead":

```console
$ curl -siH 'Accept: application/vnd.docker.distribution.manifest.list.v2+json' https://quay.io/v2/openshift-release-dev/ocp-release/manifests/4.15.5-s390x
HTTP/2 200 
date: Tue, 27 May 2025 22:48:02 GMT
content-type: application/vnd.docker.distribution.manifest.v1+json    <-- not a manifest-list, even though we gave manifest-list as our preferred Accept media-type
...

Reply:
Done.